command_test.go

```diff
@@ -351,6 +351,121 @@ func TestCommand_Run_BeforeReturnNewContext(t *testing.T) {
 	require.Equal(t, "bval", receivedValFromAction)
 }
 
+func TestCommand_Run_BeforeReturnNewContextSubcommand(t *testing.T) {
+	type key string
+
+	bkey := key("bkey")
+	bkey2 := key("bkey2")
+
+	contextValues := make(map[string]string)
```

Comment:
```suggestion
	contextValues := make(map[string]map[string]string)
```

Make it a map from string(function name) to dict